text/0016-standard-modes.md

```diff
@@ -0,0 +1,306 @@
+
+This is therefore a big step towards MAVLink being capable of being truly autopilot-agnostic for a useful minimal set of functionality.
+
+
+### Background
+
+MAVLink currently standardizes *very high level* _base modes_ (defined in [MAV_MODE](https://mavlink.io/en/messages/common.html#MAV_MODE)), which cover things like whether the vehicle is ready to fly, being tested, manually controlled, under GCS manual control, or executing a mission.
+
+More specific flight behaviour/modes are defined in _custom modes_, which are specific to each flight stack.
+Mechanisms are provided to set the base and custom modes ([MAV_CMD_DO_SET_MODE](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_SET_MODE)).
+The `HEARTBEAT` contains the current/active `base_mode` and `custom_mode`.
+There are no mechanisms query the available custom modes.
+
+Most autopilots implement a set of custom flight modes that have very similar behaviour or intent, for example: position-hold mode, altitude-hold mode, mission mode, safety return/RTL.
+
+However because these are all identified by different custom mode identifiers on different flight stacks, there is no way to be sure what these "mean" without pre-existing knowledge.
+
+
+# Detailed Design
+
+The design proposes a small set of modes that are (broadly speaking) commonly implemented across flight stacks, along with mechanisms to:
+- determine what modes are available
+- set the standard mode
+- infer the current standard mode, if active.
+
+
+## Standard modes
+
+Standard modes are those that are implemented in a similar way by most flight stacks, and which a GCS can present as more or less the same thing for any flight stack.
+
+The precise mechanics and behaviour do not have to be identical, but the broad intent has to be similar, and any differing behaviour across flight statcks should not be unexpected by a user.
+
+A flight stack is not _required_ to support all of these modes.
+
+The standard modes will be defined in an enum `MAV_STANDARD_MODE`, which has an initial value `0` is an enum that indicates "not a standard mode".
+
+The proposed initial set of modes is:
+```xml
+    <enum name="MAV_STANDARD_MODE">
+      <description>Standard modes with a well understood meaning across flight stacks and vehicle types.
+        For example, most flight stack have the concept of a "return" or "RTL" mode that takes a vehicle to safety, even though the precise mechanics of this mode may differ.
+        Modes may be set using MAV_CMD_DO_SET_STANDARD_MODE.
+      </description>
+      <entry value="0" name="MAV_STANDARD_MODE_NON_STANDARD">
+        <description>Non standard mode.
+          This may be used when reporting the mode if the current flight mode is not a standard mode.
+        </description>
+      </entry>
+      <entry value="1" name="MAV_STANDARD_MODE_POSITION_HOLD">
```

Comment:
Discussion on the standards call: It would make sense to add an orbit mode and a way to shift the orbit center.

Reply:
Resolved, needs review.